src/util/TypeGuards.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const { ApplicationCommandOptionTypes } = require('./Constants');
+
+/**
+ * Helpers for checking different variants of common types.
+ */
+class TypeGuards extends null {
+  /**
+   * Resolves a given type to an enum equivalent value, and
+   * checks if it's part of the given enum type.
+   * @private
+   * @param {string|number} type The type to resolve
+   * @param {any} object The enum to resolve to
+   * @param {string[]} fields The enum fields to check
+   * @returns {boolean} Whether the type is part of the enum
+   */
+  static isPartOfEnum(type, object, fields) {
+    const resolvedType = typeof type === 'number' ? type : object[type];
+    return fields.some(field => object[field] === resolvedType);
+  }
```

Comment:
Should probably have some more checks to make sure all parameters are of the desired types and throw errors if not

Reply:
This is private, it's only meant for use within the `TypeGuards` class.

Reply:
So? This is used by the other methods which can still receive invalid types so it’s best to have checks for them here

Reply:
JS already throws an error on its own if you mess a parameter up. I don't see the value of adding another custom one. This isn't typescript, it's javascript where errors like that are normal.

Reply:
But it’s a less insightful error and the library throws TypeErrors all the time so I don’t see why this shouldn’t either

Reply:
Because as I said earlier it's a private function, any error that is thrown by that function can only be caused by the d.js devs themselves, not the public.

Reply:
That is not true because you’re passing the parameters sent by the user to the function so it can error out

Reply:
Then I can throw errors in the public functions themselves? Throwing them in the private function makes them less insightful? By the time it reaches the private function we don't know what specific type was trying to be checked.

Reply:
Sure, I just think they'd be the same so they could all be put in the same place but you do you